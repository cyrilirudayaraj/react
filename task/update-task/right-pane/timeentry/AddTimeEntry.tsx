import React, { Component } from 'react';
import {
  Button,
  Lightbox,
  Form,
  FormField,
  Select,
  DateInput,
} from '@athena/forge';
import { ButtonVariant } from '@athena/forge/Button/Button';
import { Formik, FormikProps } from 'formik';
import * as Yup from 'yup';
import { TaskStep } from '../../../../../types';
import Labels from '../../../../../constants/Labels';
import { getTaskDetail } from '../../../../../slices/TaskSlice';
import { connect } from 'react-redux';
import { fetchTaskDetails } from '../../../../../slices/TaskSlice';
import AppConstants from '../../../../../constants/AppConstants';
import ConversionUtil from '../../../../../utils/ConversionUtil';
import CommonUtil from '../../../../../utils/CommonUtil';

interface AddTimeEntryProps {
  onConfirm: (values: any) => any;
  className: string;
  id: string;
  headerText: string;
  context: string;
  icon: string;
  variant: ButtonVariant;
  disabled: boolean;
  text?: string;
  timeentry?: any;
  type?: string[];
  activeWorkflowStepId?: any;
  task: any;
  fetchTaskDetails?: (value: any) => any;
}

export class AddTimeEntry extends Component<AddTimeEntryProps, any> {
  state = {
    shown: false,
  };
  constructor(props: AddTimeEntryProps) {
    super(props);
  }

  showLightbox = (formik: FormikProps<any>) => {
    this.setState({
      shown: true,
    });
    if (this.props.context === Labels.WORKLOG.CONTEXT_EDIT) {
      formik.setValues({
        spentTimeInMins: ConversionUtil.convertMinsToHm(
          this.props.timeentry?.spentTimeInMins
        ),
        spentOn: this.props.timeentry?.spentOn,
        workflowStepId: this.props.timeentry?.workflowStepId,
      });
    } else if (this.props.context === Labels.WORKLOG.CONTEXT_ADD) {
      formik.resetForm();
      formik.setValues({
        spentTimeInMins: '',
        spentOn: new Date(),
        workflowStepId: this.props.activeWorkflowStepId,
      });
    }
  };

  hideLightbox = (): void => {
    this.setState({
      shown: false,
    });
  };

  handleWorkflowStepIdChange = (event: any, formik: any): void => {
    formik.setFieldValue(event.target.id, event.target.value);
  };

  initialValues = {
    spentTimeInMins: this.props.timeentry?.spentTimeInMins || '',
    spentOn: this.props.timeentry?.spentOn || new Date(),
    workflowStepId:
      this.props.timeentry?.workflowStepId || this.props.activeWorkflowStepId,
  };

  convertDHMToMinutes(value: string) {
    const splitArr = value.split(' ');
    let initialMins = 0;
    for (let i = 0; i < splitArr.length; i++) {
      if (splitArr[i].includes('d')) {
        initialMins =
          initialMins +
          parseInt(splitArr[i].slice(0, splitArr[i].indexOf('d'))) * 1440;
      }
      if (splitArr[i].includes('h')) {
        initialMins =
          initialMins +
          parseInt(splitArr[i].slice(0, splitArr[i].indexOf('h'))) * 60;
      }
      if (splitArr[i].includes('m')) {
        initialMins =
          initialMins +
          parseInt(splitArr[i].slice(0, splitArr[i].indexOf('h')));
      }
    }
    return initialMins;
  }

  isValidTimeFomat(value: string): boolean {
    return AppConstants.UI_CONSTANTS.TIME_FORMAT_REGEX.test(value.trim());
  }

  validationSchema = Yup.object().shape({
    spentTimeInMins: Yup.string()
      .test('spent-time', Labels.WORKLOG.SPENT_ON_VALIDATION_TEXT, (value) =>
        this.isValidTimeFomat(value)
      )
      .required('Required!'),
    spentOn: Yup.string().required('Required!'),
    workflowStepId: Yup.string().required('Required!'),
  });

  getFormFieldProps = (fieldName: string, formik: any) => {
    const props = {
      ...formik.getFieldProps(fieldName),
      error: formik.errors[fieldName],
      id: fieldName,
      labelWidth: 12,
      required: true,
    };
    return props;
  };

  getWorkflowStepOrdering(activeWorkflowStepId: any): number {
    const activeTaskStep = this.props.task?.taskSteps?.find(
      (taskStep: TaskStep) => taskStep.workflowStepId === activeWorkflowStepId
    );
    return parseInt(activeTaskStep?.ordering);
  }

  getPreviousWorkflowStepOptions(
    taskSteps: any,
    workflowStepId: string
  ): any[] {
    const ordering = this.getWorkflowStepOrdering(workflowStepId);
    return taskSteps
      .filter((taskStep: TaskStep) => parseInt(taskStep.ordering) <= ordering)
      .map((ts: TaskStep) => {
        return { text: ts.name, value: ts.workflowStepId };
      });
  }

  getWorkLogWorkflowStepOptions(): any[] {
    const { taskSteps } = this.props.task;
    let convertedArr = [];
    if (this.props.context === Labels.WORKLOG.CONTEXT_ADD) {
      convertedArr = this.getPreviousWorkflowStepOptions(
        taskSteps,
        this.props.activeWorkflowStepId
      );
    } else if (this.props.context === Labels.WORKLOG.CONTEXT_EDIT) {
      convertedArr = this.getPreviousWorkflowStepOptions(
        taskSteps,
        this.props.timeentry.workflowStepId
      );
    }
    return convertedArr;
  }

  renderForm = (formik: FormikProps<any>) => {
    return (
      <Form
        labelAlwaysAbove={true}
        includeSubmitButton={false}
        requiredVariation="blueBarWithLegend"
        className="fe_u_margin--large"
        layout="compact"
      >
        <FormField
          labelText="Time Spent"
          {...this.getFormFieldProps('spentTimeInMins', formik)}
          hintText={Labels.WORKLOG.SPENT_ON_HINT_TEXT}
        />

        <FormField
          labelText="Workflow Step"
          inputAs={Select}
          options={this.getWorkLogWorkflowStepOptions()}
          {...this.getFormFieldProps('workflowStepId', formik)}
          onChange={(event: any) => {
            this.handleWorkflowStepIdChange(event, formik);
          }}
        />
        <FormField
          labelText="Date Started"
          inputAs={DateInput}
          required
          minDate={new Date('01-01-1900')}
          maxDate={new Date()}
          {...this.getFormFieldProps('spentOn', formik)}
          onBlur={(event: any) => {
            CommonUtil.handleDateChange(event, formik);
          }}
        />
      </Form>
    );
  };

  render(): JSX.Element {
    return (
      <div className={this.props.className}>
        <Formik
          initialValues={this.initialValues}
          validationSchema={this.validationSchema}
          onSubmit={(values) => {
            this.props.onConfirm({
              ...values,
              spentTimeInMins: this.convertDHMToMinutes(values.spentTimeInMins),
              id: this.props?.timeentry?.id,
              version: this.props?.timeentry?.version,
            });
            this.hideLightbox();
          }}
          validateOnChange={false}
          validateOnBlur={false}
          validateOnMount={false}
        >
          {(formik) => {
            return (
              <>
                <Button
                  className={this.props.className}
                  id={this.props.id}
                  text={this.props.text}
                  icon={this.props.icon}
                  variant={this.props.variant}
                  disabled={this.props.disabled}
                  onClick={() => this.showLightbox(formik)}
                />
                <Lightbox
                  show={this.state.shown}
                  hideDividers
                  headerText={this.props.headerText}
                  disableClose
                  width="large"
                  className="my-custom-lightbox create_document"
                >
                  <Button
                    variant="tertiary"
                    icon={Labels.WORKLOG.ICON_CLOSE}
                    onClick={() => this.hideLightbox()}
                    className="my-close-button"
                  />
                  {this.renderForm(formik)}

                  <div className="fe_c_lightbox__footer">
                    <Button
                      text={Labels.WORKLOG.BUTTON_CANCEL}
                      variant="secondary"
                      className="fe_u_margin--right-small"
                      onClick={() => this.hideLightbox()}
                    />
                    <Button
                      text={this.props.context}
                      onClick={(values) => {
                        formik.handleSubmit(values);
                      }}
                    />
                  </div>
                </Lightbox>
              </>
            );
          }}
        </Formik>
      </div>
    );
  }
}

const mapTaskStateToProps = (state: any) => ({
  task: getTaskDetail(state),
});

export default connect(mapTaskStateToProps, { fetchTaskDetails })(AddTimeEntry);
