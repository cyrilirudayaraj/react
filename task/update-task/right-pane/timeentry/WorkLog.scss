.work-log {
  width: 100%;
  max-height: 100%;
  margin: auto;
  overflow: hidden;
  .worklog-header {
    display: flex;
    justify-content: space-between;
    box-shadow: 0 1.5px 0px 0 #cdcdcd;
    .add-button {
      button {
        padding-bottom: 5px;
      }
    }
    .add-button.fe_c_button:focus {
      box-shadow: none;
    }
    .worklog-time-container {
      .time-label {
        color: #333333;
        font-family: 'Source Sans Pro';
        font-size: 16px;
        letter-spacing: 0;
        line-height: 24px;
        text-align: right;
      }
      .document-type {
        color: #333333;
        font-family: 'Source Sans Pro';
        font-size: 16px;
        font-weight: 600;
        letter-spacing: 0;
        line-height: 24px;
      }
    }
  }

  .worklog-list-container {
    scrollbar-width: thin;
    max-height: 65vh;
    font-family: 'Source Sans Pro';
    font-size: 14px;
    letter-spacing: 0;
    line-height: 18px;
    overflow-y: auto;
    display: flex;
    flex-direction: column-reverse;
    margin-bottom: 20px;
    .worklog-list-group {
      width: 100%;
      margin-bottom: 0;
      padding-bottom: 0;
      margin-left: 0;
      .fe_c_list.fe_c_accordion {
        min-width: 100%;
        border-top: 0;
        .fe_c_list__item.fe_c_accordion-item {
          min-width: 100%;
          h3 {
            color: #333433;
            font-family: 'Source Sans Pro';
            font-size: 16px;
            letter-spacing: 0px;
            line-height: 24px;
          }
          .fe_c_accordion-item__content {
            min-width: 100%;
            .fe_c_accordion-item__content-inner {
              min-width: 100%;
              .fe_c_accordion-item__padded-content {
                min-width: 100%;
                .worklog-container {
                  min-width: 100%;
                  min-height: 50px;
                  list-style: none;
                  padding-top: 7px;
                  margin-bottom: 0;
                  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
                  div {
                    display: inline-block;
                  }
                  .worklog-created-container {
                    min-width: 50%;
                    border-left: 5px solid #d9d9d9;
                    .created-by {
                      margin-top: 0;
                      color: #333333;
                      font-family: 'Source Sans Pro';
                      font-size: 16px;
                      letter-spacing: 0;
                      line-height: 21px;
                      padding-left: 5px;
                    }
                  }
                  .worklog-spenttime-container {
                    min-width: 30%;
                    .spent-time {
                      color: #333333;
                      font-family: 'Source Sans Pro';
                      font-size: 14px;
                      font-weight: 600;
                      letter-spacing: 0;
                      line-height: 21px;
                      text-align: right;
                    }
                  }

                  .edit-button {
                    display: none;
                    margin-right: 4%;
                  }
                  .delete-button {
                    display: none;
                  }
                }
                .worklog-container:hover .edit-button,
                .edit-button:hover {
                  display: inline-block;
                }
                .worklog-container:hover {
                  background-color: #d9ebfa;
                }
                .worklog-container:hover .delete-button,
                .delete-button:hover {
                  display: inline-block;
                }
              }
            }
          }
        }
      }
    }
    .worklog-list-group {
      .worklog-container:last-child {
        border-bottom: 0;
      }
    }
  }
}

.worklog-list-container::-webkit-scrollbar {
  width: 5px;
}
.worklog-list-container::-webkit-scrollbar-thumb {
  border-radius: 10px;
}
