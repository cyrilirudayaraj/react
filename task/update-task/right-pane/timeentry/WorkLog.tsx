import React, { Component } from 'react';
import { connect } from 'react-redux';
import { WorkLogProps } from '../../../../../types';
import { getTaskDetail } from '../../../../../slices/TaskSlice';
import WorkLogList from './WorkLogList';
import { fetchTimeEntryLogs } from '../../../../../slices/TimeEntrySlice';
import './WorkLog.scss';
class WorkLog extends Component<WorkLogProps, any> {
  constructor(props: WorkLogProps) {
    super(props);
    this.props.fetchTimeEntryLogs(this.props.task.id);
  }

  render(): JSX.Element {
    return (
      <div className="work-log">
        <WorkLogList />
      </div>
    );
  }
}

const mapTaskStateToProps = (state: any) => ({
  task: getTaskDetail(state),
});

export default connect(mapTaskStateToProps, {
  fetchTimeEntryLogs,
})(WorkLog);
