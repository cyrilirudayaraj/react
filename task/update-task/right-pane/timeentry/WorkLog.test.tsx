import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { TaskDetail, TaskStep } from '../../../../../types/index';
import WorkLog from './WorkLog';
import { WorkLogList } from './WorkLogList';
import { Formik } from 'formik';
import { AddTimeEntry } from './AddTimeEntry';
import { act } from 'react-dom/test-utils';
configure({ adapter: new Adapter() });

let wrapper: any;
let wrapper1: any;
let wrapperConnector: any;
let taskDetails: TaskDetail;
let store: any;
const timeEntryLogs: any = [];
let payload: any;
let deletePayload: any;
beforeEach(() => {
  taskDetails = {} as TaskDetail;
  taskDetails.activeTaskStepId = '1';
  const taskStep: TaskStep = {} as TaskStep;
  taskDetails.taskSteps = [] as TaskStep[];
  taskDetails.taskSteps[0] = taskStep;
  taskDetails.taskSteps[0].name = 'Analyze Task';
  taskDetails.taskSteps[0].workflowStepId = '1';
  taskDetails.taskSteps[0].ordering = '1';
  timeEntryLogs[0] = {
    spentOn: '08/15/2020',
    createdBy: 'jcyril',
    spentTimeInMins: '1660',
    workflowStepId: '1',
    workflowStepName: 'Analyze Task',
    id: '10',
    taskId: '1',
  };
  timeEntryLogs[1] = {
    spentOn: '08/16/2020',
    createdBy: 'jcyril',
    spentTimeInMins: '1660',
    workflowStepId: '2',
    workflowStepName: 'Review Task',
    id: '11',
    taskId: '1',
  };

  store = configureStore([])({
    task: {
      taskDetails: taskDetails,
    },
    userPermission: {
      userPermissions: ['task.update'],
    },
    validation: {
      formInvalid: {
        isWorkLogFormValid: true,
      },
    },
    timeEntry: {
      timeEntryLogs: timeEntryLogs,
    },
    fetchTimeEntryLogs: jest.fn(),
    setTimeEntryLogs: jest.fn(),
  });

  store.dispatch = jest.fn();
  wrapperConnector = mount(
    <Provider store={store}>
      <Formik initialValues={{}} onSubmit={jest.fn()} validate={jest.fn()}>
        <WorkLog />
      </Formik>
    </Provider>
  );
  wrapper = wrapperConnector.find(WorkLogList);
  wrapper1 = wrapperConnector.find(AddTimeEntry);
  payload = {
    spentOn: '08/15/2020',
    createdBy: 'jcyril',
    spentTimeInMins: '30',
    workflowStepId: '1',
    workflowStepName: 'Analyze Task',
    id: '10',
    taskId: '1',
    version: '1',
    deleted: '09/16/2020',
  };
  deletePayload = {
    id: '1',
    taskId: '1',
    deleted: '09/16/2020',
    version: '1',
    workflowStepId: '1',
    spentTimeInMins: '30',
  };
});

describe('Work Log', () => {
  it('basic snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
    wrapper.instance().addWorkLog(payload);
    wrapper.instance().updateWorkLog(payload);
    wrapper.instance().deleteWorkLog('1', '1', deletePayload);
  });

  it('basic time entry snap', () => {
    expect(wrapper1.debug()).toMatchSnapshot();
  });
  it('should close the dialog when close button clicked', () => {
    act(() => {
      wrapper1.setState({ shown: true });
      wrapper1.instance().hideLightBox;
      expect(wrapper1.state().shown).toEqual(false);
    });
  });
});
