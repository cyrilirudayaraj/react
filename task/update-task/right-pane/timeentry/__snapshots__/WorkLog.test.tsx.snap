// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work Log basic snapshot 1`] = `
"<WorkLogList workLogs={{...}} taskdetails={{...}} rightPaneFormvalidity={{...}} refreshTaskList={[Function (anonymous)]} createTimeEntryLog={[Function (anonymous)]} updateTimeEntryLog={[Function (anonymous)]} deleteTimeEntryLog={[Function (anonymous)]} fetchTaskDetails={[Function (anonymous)]} updateFormValidation={[Function (anonymous)]}>
  <div className=\\"worklog-header\\">
    <Connect(WFEnableForPermission) permission=\\"worklog.create\\">
      <WFEnableForPermission permission=\\"worklog.create\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
        <Connect(AddTimeEntry) className=\\"add-button\\" id=\\"add-button\\" headerText=\\"Log Time\\" context=\\"Add\\" icon=\\"Add\\" text=\\"Log Time\\" variant=\\"tertiary\\" disabled={false} onConfirm={[Function (anonymous)]} activeWorkflowStepId={[undefined]}>
          <AddTimeEntry className=\\"add-button\\" id=\\"add-button\\" headerText=\\"Log Time\\" context=\\"Add\\" icon=\\"Add\\" text=\\"Log Time\\" variant=\\"tertiary\\" disabled={false} onConfirm={[Function (anonymous)]} activeWorkflowStepId={[undefined]} task={{...}} fetchTaskDetails={[Function (anonymous)]}>
            <div className=\\"add-button\\">
              <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: onSubmit]} validateOnChange={false} validateOnBlur={false} validateOnMount={false}>
                <ForwardRef className=\\"add-button\\" id=\\"add-button\\" text=\\"Log Time\\" icon=\\"Add\\" variant=\\"tertiary\\" disabled={false} onClick={[Function: onClick]} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                  <button id=\\"add-button\\" className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium add-button\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                    <div className=\\"fe_c_button__content\\">
                      <Add title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
                        <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
                          <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                            <g>
                              <g className=\\"fe_c_icon__background\\">
                                <rect width=\\"25\\" height=\\"25\\" />
                              </g>
                              <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                <g>
                                  <polygon points=\\"13.3 7.8 11.8 7.8 11.8 11.8 7.8 11.8 7.8 13.3 11.8 13.3 11.8 17.2 13.3 17.2 13.3 13.3 17.2 13.3 17.2 11.8 13.3 11.8 13.3 7.8\\" />
                                  <path d=\\"M20,5a11,11,0,0,0-7.5-3.1A11,11,0,0,0,5,5,10.7,10.7,0,0,0,5,20a11,11,0,0,0,7.5,3.1A11,11,0,0,0,20,20,10.7,10.7,0,0,0,20,5ZM18.9,18.9a9.1,9.1,0,0,1-6.4,2.7,9.1,9.1,0,0,1,0-18.2,9.1,9.1,0,0,1,6.4,15.5Z\\" />
                                </g>
                              </g>
                            </g>
                          </svg>
                        </IconBase>
                      </Add>
                      <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--tertiary\\">
                        Log Time
                      </span>
                    </div>
                  </button>
                </ForwardRef>
                <ThemeConsumerHOC show={false} hideDividers={true} headerText=\\"Log Time\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\">
                  <Lightbox show={false} hideDividers={true} headerText=\\"Log Time\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\" theme={{...}} closeButtonAriaLabel=\\"Close Modal\\" includeButtonRow={false} primaryButtonText=\\"Next\\" padded={true}>
                    <ModalManagerWrapper className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false}>
                      <Overlay manager={{...}} className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false} autoFocus={false} enforceFocus={false}>
                        <Modal aria-modal=\\"true\\" aria-labelledby={[undefined]} aria-describedby={[undefined]} renderBackdrop={[Function: renderBackdrop]} transition={[Function: FadeUp]} backdropTransition={[Function: Fade]} manager={{...}} onShow={[Function (anonymous)]} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false} autoFocus={false} enforceFocus={false} className=\\"fe_c_overlay fe_f_all fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" role=\\"dialog\\" backdrop={true} keyboard={true} restoreFocus={true} />
                      </Overlay>
                    </ModalManagerWrapper>
                  </Lightbox>
                </ThemeConsumerHOC>
              </Formik>
            </div>
          </AddTimeEntry>
        </Connect(AddTimeEntry)>
      </WFEnableForPermission>
    </Connect(WFEnableForPermission)>
    <div className=\\"worklog-time-container\\">
      <span className=\\"time-label\\">
        Total Logged Time : 
      </span>
      <span className=\\"document-type\\">
        55h 20m
      </span>
    </div>
  </div>
  <div className=\\"worklog-list-container\\">
    <ul className=\\"worklog-list-group\\">
      <Accordion>
        <List className=\\"fe_c_accordion\\" padded={false} dividers={false}>
          <ul className=\\"fe_c_list fe_c_accordion\\">
            <AccordionItem headingText=\\"Analyze Task\\" defaultExpanded={false} headerSlot=\\"27h 40m\\" headingTag=\\"h3\\" mountedWhileHidden={false} padded={true} tabIndex={0}>
              <ListItem className=\\"fe_c_accordion-item\\">
                <li className=\\"fe_c_list__item fe_c_accordion-item\\">
                  <div className=\\"fe_c_accordion-item__header\\" onClick={[Function: bound toggleExpand]} onKeyDown={[Function: onKeyDown]} aria-expanded={false} role=\\"button\\" tabIndex={0}>
                    <Icon icon=\\"Expand\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\" iconSet={{...}}>
                      <Expand title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                        <IconBase viewBox=\\"0 0 25 25\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                          <svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_icon fe_c_accordion-item__expand\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                            <title>
                              Expand
                            </title>
                            <g>
                              <g className=\\"fe_c_icon__background\\">
                                <rect width=\\"25\\" height=\\"25\\" />
                              </g>
                              <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                <polygon points=\\"12.5 16.6 6 10 7 9 12.5 14.4 18 9 19 10 12.5 16.6\\" />
                              </g>
                            </g>
                          </svg>
                        </IconBase>
                      </Expand>
                    </Icon>
                    <Heading text=\\"Analyze Task\\" variant=\\"section\\" headingTag=\\"h3\\" className=\\"fe_c_accordion-item__heading-text\\">
                      <h3 className=\\"fe_c_heading fe_c_heading--section fe_c_accordion-item__heading-text\\">
                        Analyze Task
                      </h3>
                    </Heading>
                    <div className=\\"fe_c_accordion-item__header-slot\\">
                      27h 40m
                    </div>
                  </div>
                  <div className=\\"fe_c_accordion-item__content fe_is-exited\\">
                    <CSSTransition in={false} classNames=\\"fe_is-expanding\\" timeout={400} onEnter={[Function: bound handleEnter]} onEntering={[Function: handleEntering]} onEntered={[Function: handleEntered]} onExit={[Function: handleExit]} onExiting={[Function: handleExiting]} onExited={[Function: bound handleExited]} mountOnEnter={true} unmountOnExit={true}>
                      <Transition in={false} timeout={400} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={true} unmountOnExit={true} appear={false} enter={true} exit={true} />
                    </CSSTransition>
                  </div>
                </li>
              </ListItem>
            </AccordionItem>
          </ul>
        </List>
      </Accordion>
    </ul>
  </div>
</WorkLogList>"
`;

exports[`Work Log basic time entry snap 1`] = `
"<AddTimeEntry className=\\"add-button\\" id=\\"add-button\\" headerText=\\"Log Time\\" context=\\"Add\\" icon=\\"Add\\" text=\\"Log Time\\" variant=\\"tertiary\\" disabled={false} onConfirm={[Function (anonymous)]} activeWorkflowStepId={[undefined]} task={{...}} fetchTaskDetails={[Function (anonymous)]}>
  <div className=\\"add-button\\">
    <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: onSubmit]} validateOnChange={false} validateOnBlur={false} validateOnMount={false}>
      <ForwardRef className=\\"add-button\\" id=\\"add-button\\" text=\\"Log Time\\" icon=\\"Add\\" variant=\\"tertiary\\" disabled={false} onClick={[Function: onClick]} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
        <button id=\\"add-button\\" className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium add-button\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
          <div className=\\"fe_c_button__content\\">
            <Add title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
              <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                  <g>
                    <g className=\\"fe_c_icon__background\\">
                      <rect width=\\"25\\" height=\\"25\\" />
                    </g>
                    <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                      <g>
                        <polygon points=\\"13.3 7.8 11.8 7.8 11.8 11.8 7.8 11.8 7.8 13.3 11.8 13.3 11.8 17.2 13.3 17.2 13.3 13.3 17.2 13.3 17.2 11.8 13.3 11.8 13.3 7.8\\" />
                        <path d=\\"M20,5a11,11,0,0,0-7.5-3.1A11,11,0,0,0,5,5,10.7,10.7,0,0,0,5,20a11,11,0,0,0,7.5,3.1A11,11,0,0,0,20,20,10.7,10.7,0,0,0,20,5ZM18.9,18.9a9.1,9.1,0,0,1-6.4,2.7,9.1,9.1,0,0,1,0-18.2,9.1,9.1,0,0,1,6.4,15.5Z\\" />
                      </g>
                    </g>
                  </g>
                </svg>
              </IconBase>
            </Add>
            <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--tertiary\\">
              Log Time
            </span>
          </div>
        </button>
      </ForwardRef>
      <ThemeConsumerHOC show={false} hideDividers={true} headerText=\\"Log Time\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\">
        <Lightbox show={false} hideDividers={true} headerText=\\"Log Time\\" disableClose={true} width=\\"large\\" className=\\"my-custom-lightbox create_document\\" theme={{...}} closeButtonAriaLabel=\\"Close Modal\\" includeButtonRow={false} primaryButtonText=\\"Next\\" padded={true}>
          <ModalManagerWrapper className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false}>
            <Overlay manager={{...}} className=\\"fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" dialogClass={{...}} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false} autoFocus={false} enforceFocus={false}>
              <Modal aria-modal=\\"true\\" aria-labelledby={[undefined]} aria-describedby={[undefined]} renderBackdrop={[Function: renderBackdrop]} transition={[Function: FadeUp]} backdropTransition={[Function: Fade]} manager={{...}} onShow={[Function (anonymous)]} onHide={[Function: handleOnHide]} aria-label=\\"Log Time\\" show={false} autoFocus={false} enforceFocus={false} className=\\"fe_c_overlay fe_f_all fe_c_lightbox fe_f_all my-custom-lightbox create_document\\" role=\\"dialog\\" backdrop={true} keyboard={true} restoreFocus={true} />
            </Overlay>
          </ModalManagerWrapper>
        </Lightbox>
      </ThemeConsumerHOC>
    </Formik>
  </div>
</AddTimeEntry>"
`;
